### PR TITLE
chore: Add PHONY to Makefile targets

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

**Comment by karlkfi** [comment]
Looks like a flaky test. Fix: https://github.com/kubernetes-sigs/cli-utils/pull/581

File: Makefile [hunk 1]
```diff
@@ -35,39 +39,47 @@ install-column-apt:
...
+	go install github.com/google/addlicense@v1.0.0
 
-install-lint:
-	(which $(GOPATH)/bin/golangci-lint || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.44.0)
+$(MYGOBIN)/golangci-lint:
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.44.0
 
-install-deepcopy-gen:
+$(MYGOBIN)/deepcopy-gen:
 	(which $(GOPATH)/bin/deepcopy-gen || go install k8s.io/code-generator/cmd/deepcopy-gen@v0.23.3)
```
Comment (ash2k) [inline Makefile:52]: Forgot to remove `which`?

File: Makefile [hunk 2]
```diff
@@ -35,39 +39,47 @@ install-column-apt:
...
 
-generate: install-stringer generate-deepcopy
+.PHONY: generate
+generate: $(MYGOBIN)/stringer generate-deepcopy
 	go generate ./...
 
-license: install-addlicense
+.PHONY: license
+license: $(MYGOBIN)/addlicense
 	$(GOPATH)/bin/addlicense -v -y 2021 -c "The Kubernetes Authors." -f LICENSE_TEMPLATE .
```
Comment (ash2k) [inline Makefile:64]:
```suggestion
	$(MYGOBIN)/addlicense -v -y 2021 -c "The Kubernetes Authors." -f LICENSE_TEMPLATE .
```

File: Makefile [hunk 3]
```diff
@@ -35,39 +39,47 @@ install-column-apt:
...
 
-verify-license: install-addlicense
+.PHONY: verify-license
+verify-license: $(MYGOBIN)/addlicense
 	$(GOPATH)/bin/addlicense  -check .
```
Comment (ash2k) [inline Makefile:68]:
```suggestion
	$(MYGOBIN)/addlicense  -check .
```

File: Makefile [hunk 4]
```diff
@@ -35,39 +39,47 @@ install-column-apt:
...
 
+.PHONY: tidy
 tidy:
 	go mod tidy
 
-lint: install-lint
+.PHONY: lint
+lint: $(MYGOBIN)/golangci-lint
 	$(GOPATH)/bin/golangci-lint run ./...
```
Comment (ash2k) [inline Makefile:76]:
```suggestion
	$(MYGOBIN)/golangci-lint run ./...
```

File: Makefile [hunk 5]
```diff
@@ -77,17 +89,21 @@ test-e2e-focus: $(MYGOBIN)/ginkgo $(MYGOBIN)/kind
 	kind delete cluster --name=cli-utils-e2e && kind create cluster --name=cli-utils-e2e --wait 5m
 	$(GOPATH)/bin/ginkgo -v -focus ".*$(FOCUS).*" ./test/e2e/... -- -v 5
 
+.PHONY: test-stress
 test-stress: $(MYGOBIN)/ginkgo $(MYGOBIN)/kind
 	kind delete cluster --name=cli-utils-e2e && kind create cluster --name=cli-utils-e2e --wait 5m
 	$(GOPATH)/bin/ginkgo -v ./test/stress/... -- -v 3
```
Comment (ash2k) [inline Makefile:95]:
```suggestion
	$(MYGOBIN)/ginkgo -v ./test/stress/... -- -v 3
```

File: Makefile [hunk 6]
```diff
@@ -77,17 +89,21 @@ test-e2e-focus: $(MYGOBIN)/ginkgo $(MYGOBIN)/kind
 	kind delete cluster --name=cli-utils-e2e && kind create cluster --name=cli-utils-e2e --wait 5m
 	$(GOPATH)/bin/ginkgo -v -focus ".*$(FOCUS).*" ./test/e2e/... -- -v 5
```
Comment (ash2k) [inline Makefile:90]:
```suggestion
	$(MYGOBIN)/ginkgo -v -focus ".*$(FOCUS).*" ./test/e2e/... -- -v 5
```

File: Makefile [hunk 7]
```diff
@@ -35,39 +39,47 @@ install-column-apt:
...
 	$(GOPATH)/bin/golangci-lint run ./...
 
+.PHONY: test
 test:
 	go test -race -cover ./cmd/... ./pkg/...
 
+.PHONY: test-e2e
 test-e2e: $(MYGOBIN)/ginkgo $(MYGOBIN)/kind
 	kind delete cluster --name=cli-utils-e2e && kind create cluster --name=cli-utils-e2e --wait 5m
 	$(GOPATH)/bin/ginkgo -v ./test/e2e/... -- -v 3
```
Comment (ash2k) [inline Makefile:85]:
```suggestion
	$(MYGOBIN)/ginkgo -v ./test/e2e/... -- -v 3
```

**Review by ash2k — COMMENTED** [review]
Nice cleanup.

1. If `GOPATH` contains a path with a space, it will be used to split the path into pieces where it's not double quoted, right? I think the right thing to do would be to quote all `$(MOGOBIN)` and `$(GOPATH)` references.
2. `GOPATH` is a list of paths, not a single path. This is annoying. I'm not asking to fix how things work with it as it's very rare in practice, just FYI :)

File: Makefile (hunk 1 above)
Comment (karlkfi) [reply Makefile:52]: oops, yeah. will fix.

**Comment by karlkfi** [comment]
> 1. If `GOPATH` contains a path with a space, it will be used to split the path into pieces where it's not double quoted, right? I think the right thing to do would be to quote all `$(MOGOBIN)` and `$(GOPATH)` references.
> 2. `GOPATH` is a list of paths, not a single path. This is annoying. I'm not asking to fix how things work with it as it's very rare in practice, just FYI :)

Oh man... it's worse than I expected...

> The GOPATH environment variable lists places to look for Go code.
> On Unix, the value is a colon-separated string.
> On Windows, the value is a semicolon-separated string.
> On Plan 9, the value is a list.

I'm not gonna parse that. If you want to use something specific, you can specify the GOPATH or MYGOBIN make var values.

I did change it to quote the value tho, so it will cause a path not found error if it's a list.

**Review by ash2k — APPROVED** [review]
/lgtm

**Review by mortent — APPROVED** [review]
/approve

**Pull request merged**